source/kraken/worker.cpp

```diff
@@ -832,15 +844,21 @@ void Worker::heat_map(const pbnavitia::HeatMapRequest& request) {
         return;
     }
 
-    type::EntryPoint ep = arg.origins.empty() ? arg.destinations[0] : arg.origins[0];
+    // If we have a center
+    // Origins or destinations are already computed stop_points
+    const auto& center =
+        arg.isochrone_center ? *arg.isochrone_center : (arg.origins.empty() ? arg.destinations[0] : arg.origins[0]);
+    const auto& stop_points = arg.isochrone_center ? (arg.origins.empty() ? arg.destinations : arg.origins)
+                                                   : boost::optional<const std::vector<type::EntryPoint>&>{};
```

Comment:
code duplication spotted 3 times :robot: 

Also, why returning a `boost<optional>` of an empty vector ? Would `boost::none` not be better in this case ?

Reply:
I refactored that part in a commit but broke the CI with really ugly code, I am working on that right now.

I used a `boost<optional>` of an empty vector because auto can't resolve the type in a ternary if : `arg.origins` and `arg.destinations` are not the same type of `boost::none`.

I think of using something like `    const auto& stop_points =
        boost::make_optional(arg.isochrone_center, (arg.origins.empty() ? arg.destinations : arg.origins));`